files/file.go

```diff
@@ -175,7 +163,7 @@ func (i *FileInfo) detectType(modify, saveContent bool) error {
 	case strings.HasPrefix(mimetype, "image"):
 		i.Type = "image"
 		return nil
-	case isBinary(buffer[:n], n) || i.Size > 10*1024*1024: // 10 MB
+	case (len(buffer) > 0 && isBinary(buffer)) || i.Size > 10*1024*1024: // 10 MB
```

Comment:
If the file has known extension and it is not video,audio or image and the size is  < 10mb, it will be considered as a text file which is not correct. The logic must be changed to `strings.HasPrefix(mimetype, "text")` with `blob` as a default type

Reply:
Hey, sorry for the delay! I just updated the PR with your suggestion 🙂